gem/lib/metasploit-payloads.rb

```diff
@@ -10,48 +12,126 @@ module MetasploitPayloads
   METERPRETER_SUBFOLDER = 'meterpreter'
   USER_DATA_SUBFOLDER   = 'payloads'
 
+  #
+  # @return [Array<Hash<String, Symbol>>] An array of filenames with warnings. Provides a file name and reason.
+  #  Empty if all needed Meterpreter files exist and have the correct hash.
+  def self.file_warnings
```

Comment:
Maybe a better name? 🤔 

```suggestion
  def self.manifest_errors
```

Rationale being, the caller should decide what's a warning/error/etc